### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'sub')

### ISSUE THREAD

**Issue by MarcGodard** [feathersjs/feathers#3266]
### Steps to reproduce

- Create new project (then add authentication as per docs), setup google or github (probably others too) for oauth
- visit correct /oauth/google folder
- authorize the app
- app redirects
- get the following error:

```
error: TypeError: Cannot read properties of undefined (reading 'sub')
    at GoogleStrategy.getEntityQuery (/home/marcgodard/Documents/Github/gozamm-api/node_modules/@feathersjs/authentication-oauth/lib/strategy.js:29:41)
    at GoogleStrategy.findEntity (/home/marcgodard/Documents/Github/gozamm-api/node_modules/@feathersjs/authentication-oauth/lib/strategy.js:79:34)
    at GoogleStrategy.authenticate (/home/marcgodard/Documents/Github/gozamm-api/node_modules/@feathersjs/authentication-oauth/lib/strategy.js:118:44)
```

### Expected behavior

No error and user created

### Actual behavior

Above Error

### System configuration

Brand new feathers 5.0.8 project (javascript, npm, json schemas)

**Module versions** (especially the part that's not working): 5.0.8

**NodeJS version**: v16

**Operating System**: linux

**Comment by MarcGodard** [comment]
In `Strategy.js` Changing this:

```
    async getEntityQuery(profile, _params) {
        console.log(profile)
        return {
            [`${this.name}Id`]: profile.sub || profile.id
        };
    }
    async getEntityData(profile, _existingEntity, _params) {
        console.log(profile)
        return {
            [`${this.name}Id`]: profile.sub || profile.id
        };
    }
```

To this (in my node_module files):

```
    async getEntityQuery(profile, _params) {
        console.log(profile)
        return {
            [`${this.name}Id`]: profile?.sub ?? String(profile.id)
        };
    }
    async getEntityData(profile, _existingEntity, _params) {
        console.log(profile)
        return {
            [`${this.name}Id`]: profile?.sub ?? String(profile.id)
        };
    }
```

Made it work for both Google and Github.

**Comment by JonasTheGamer** [comment]
Hm, after applying your fix I'm now getting this error:
```
info: Feathers app listening on http://FlexibeleJonas2:3030
undefined
error: TypeError: Cannot read properties of undefined (reading 'id')
    at GitHubStrategy.getEntityQuery (C:\1_LOKALE_PROJECTEN\BloxTech\ResearchAndDevelopment\Feathersjs\feathers-chat\node_modules\@feathersjs\authentication-oauth\lib\strategy.js:30:64)
    at GitHubStrategy.findEntity (C:\1_LOKALE_PROJECTEN\BloxTech\ResearchAndDevelopment\Feathersjs\feathers-chat\node_modules\@feathersjs\authentication-oauth\lib\strategy.js:81:34)
    at GitHubStrategy.authenticate (C:\1_LOKALE_PROJECTEN\BloxTech\ResearchAndDevelopment\Feathersjs\feathers-chat\node_modules\@feathersjs\authentication-oauth\lib\strategy.js:120:44)
    ```

**Comment by MarcGodard** [comment]
@JonasTheGamer Not sure why you are getting this, other than you are not requesting that information? Is this with google or github? I didn't test other services, as I thought this was close enough to the original not to impact others.

**Comment by gypsicoder** [comment]
@MarcGodard 
I had get the same error. I think you didn't started the Database server. Please start the database server and then try again. I think the error description is wrong.

**Comment by MarcGodard** [comment]
@gypsicoder I don't think the database had anything to do with it.

**Comment by daffl** [comment]
It looks like it is a bug, there are more details reported in https://github.com/feathersjs/feathers/issues/3275 - basically it seems like an error happened (which is why there isn't a profile) but went through the process normally. Needs some more investigation but should be fixed in the next release.

**Comment by daffl** [comment]
This should be fixed via #3284 in v5.0.9 and propagate the error that likely happened here (instead of just returning an undefined profile).

**Comment by chrisbag** [comment]
I keep having this issue on v5 (no problem on v4) and I am not sure if it is because of my config or still a bug from feathers oauth.
Profile data is empty and the authentication method does not return the google token while no oauth error is shown
